### ISSUE TITLE
Is  "not create a %_sourcedir / %_specdir" really necessary?

### ISSUE THREAD

**Issue by falu0** [rpm-software-management/rpm#1408]
Refers to :
https://github.com/rpm-software-management/rpm/commit/a55cb376fa9d9f785081fa13802edc3743a879ca , https://github.com/rpm-software-management/rpm/commit/ca26c221e5d92b4908c5a07d2bba36f80379b43b (the latter not really, but they come together)

This triggers a change of behavior in rpmbuild which will break many long-existing workflow as soon as rpmbuild v4.16.0-release is adopted. Such change is already out in the fedora-33 rpm provided package (from which we have been bitten).

I understand the ratio behind it, but, if not reverted, I'd advise to communicate it better to the users, both in 
the changelog https://rpm.org/wiki/Releases/4.16.0 and/or through some significant error message.

Currently (in both 4.16.0-release and master branch), if no `%_sourcedir` path is pre-existing, you only see a `__CURL` error which doesn't really add any information from the rpm(build) POV. 

```
# rpmbuild -v -ba --undefine=_disable_source_fetch my.spec
warning: Downloading <https://some.tar.gz> to {%_topdir}/SOURCES/<some.tar.gz>
curl: (23) Failure writing output to destination
error: Couldn't download <https://some.tar.gz>
```

Reverting https://github.com/rpm-software-management/rpm/commit/a55cb376fa9d9f785081fa13802edc3743a879ca in 4.16.0-release (or the equivalent https://github.com/rpm-software-management/rpm/commit/acf5e00281d73a2f8034091241c7b0e2ba00e383 in master), looks to be sufficient (I just recompiled-installed-run the above command tho), but you may want to consider a better solution.

**Comment by davidnich** [comment]
commit a55cb376fa9d9f785081fa13802edc3743a879ca also breaks `rpmbuild -ts` for me - it creates a source RPM with an invalid spec file name:
```
dnichols@manatee:~/src/qore/git/qore$ rpm -ql ~/rpmbuild/SRPMS/qore-0.9.6-1.fc33.src.rpm 
qore-0.9.6.tar.bz2
rpm-tmp.xAdiY8
```

as the code to rename the temporary file `rpm-tmp.zAdiY8` in the output above) was removed in that commit - this of course breaks the SRPMS.

This is how the source RPM looked in f32:
```
dnichols@manatee:~/src/qore/git/qore$ rpm -ql ~/rpmbuild/SRPMS/qore-0.9.6-1.fc32.src.rpm 
qore-0.9.6.tar.bz2
qore.spec
```

**Comment by pmatilai** [comment]
A temporary file name ending up in the package is of course not intentional.

There was a reason for the change and that hasn't gone anywhere. Source fetch is not a supported feature, really. It should've been removed long ago as it's totally insecure in the general case.

**Comment by davidnich** [comment]
> A temporary file name ending up in the package is of course not intentional.

Should I open a new issue then for the rpmbuild breakage, or is this one OK to repurpose for that?  The source fetch issue brought me here only because it was caused by the same commit - it doesn't affect me personally, but the rpmbuild breakage could theoretically affect a lot of users I believe.

Thanks for your attention to this issue!

**Comment by pmatilai** [comment]
The temporary file name was fixed in #1453 and >= 4.16.1 so I think we can close this now. Thanks for reporting!
